src/settings_gui.cpp

```diff
@@ -92,24 +92,21 @@ static uint GetCurrentResolutionIndex()
 static void ShowCustCurrency();
 
 /** Window for displaying the textfile of a BaseSet. */
-template <class TBaseSet>
 struct BaseSetTextfileWindow : public TextfileWindow {
-	const TBaseSet *baseset; ///< View the textfile of this BaseSet.
+	const std::string &name; ///< Name of the content.
```

Comment:
This brings back memories to https://weblogs.openttd.org/openttd/2024/02/08.html#173319-220.
I'm not sure whether it's exactly the same issue, but it feels close enough.

Reply:
Not sure what you are saying here.

Do you mean it should just store it by value to avoid the pitfalls of using a reference...?